ros/src/tl_detector/tl_detector.py

```diff
@@ -68,6 +77,9 @@ def image_cb(self, msg):
             msg (Image): image from car-mounted camera
 
         """
+        if self.waypoints_tree is None:
+            return
+
         self.has_image = True
         self.camera_image = msg
         light_wp, state = self.process_traffic_lights()
```

Comment:
I see this is not something you've changed, but I wonder if it's really okay to do image processing inside a callback? What happens if the image processing is too slow, such that another image arrives before the first one finish? Is ROS okay with callbacks not returning quickly?

Reply:
I guess that's fine if a queue is configured. This way it should simply drop frames if the queue is full.